src/afrolabs/components/kafka.clj

```diff
@@ -822,13 +822,33 @@
                     (csp/>!! caught-up-ch
                              current-offsets))))))))
 
+(defn consumer:fully-current?
+  "Checks if the consumer's current offsets is equal to the end offsets.
+
+  This means there is nothing (right now) to consume in the topic.
+
+  Specify `timeout` as a java.time.Duration and `timeout-value` as a constant
+  so that you don't hang on this forever. `timeout-value` may be `:throw`, in which
+  case an exception will be thrown if the timeout expires before an offset value is
+  returned from the consumer.
+
+  Returns the current-offsets as the truthy value."
+  [consumer timeout timeout-value]
+  (let [current-offsets (consumer-current-offsets consumer :timeout-value :throw)
+        end-offsets (consumer-end-offsets consumer :timeout-value :throw)]
+    (when (and current-offsets
+               end-offsets
+               (= end-offsets current-offsets))
+      end-offsets)))
+
```

Comment:
Extract a utility method that checks if the consumer's current-offset is `==` topic's end-offset.